src/Tablebot/Utility/Discord.hs

```diff
@@ -231,10 +233,18 @@ getGuildEmoji ename gid = do
 
 -- | search through all known guilds for an emoji with that name
 findEmoji :: Text -> EnvDatabaseDiscord s (Maybe Emoji)
-findEmoji ename = fmap msum (liftDiscord readCache >>= cacheToEmoji)
+findEmoji ename = fmap msum (emojiServers >>= cacheToEmoji)
   where
-    cacheToEmoji :: Cache -> EnvDatabaseDiscord s [Maybe Emoji]
-    cacheToEmoji cache = mapM (getGuildEmoji ename) (keys $ cacheGuilds cache)
+    cacheToEmoji :: [GuildId] -> EnvDatabaseDiscord s [Maybe Emoji]
+    cacheToEmoji ids = mapM (getGuildEmoji ename) ids
+    emojiServers :: EnvDatabaseDiscord s [GuildId]
+    emojiServers = do
+      maybeServers <- liftIO $ lookupEnv "EMOJI_SERVERS"
+      case maybeServers of
+        Just x -> pure (read x)
```

Comment:
Wouldn't readMaybe be preferred here to prevent runtime errors? Although I guess it makes sense to break when the read of an environment variable fails.

Reply:
That's a reasonable point, but I think I'm fine with a bad server causing a crash - since that likely means the user typo'd a server and should fix it rather than it being ignored silently.

Reply:
Yeah, I definitely wanted it to fail loudly if it can't parse the ID, otherwise it would either never load an emoji (if I let the maybes nest) or silently let all servers be used despite the user intending it be restricted, neither of which is a desirable outcome. If I was feeling really high effort I could probably make the error message more intuitive, but given I wrote this in half an hour to fix an urgent issue, hope I can be forgiven not doing so